src/npc.cpp

```diff
@@ -60,7 +55,7 @@ bool Npc::load()
 	reset();
 
 	if (!scriptInterface) {
-		scriptInterface = new NpcScriptInterface();
+		scriptInterface.reset(new NpcScriptInterface());
```

Comment:
`scriptInterface = std::make_unique/shared<NpcScriptInterface>()`

Reply:
is there any technical difference or just preference?

Reply:
Mostly semantics, as `unique_ptr<T>::operator=(unique_ptr<T>&& rhs)` is equivalent to `lhs.reset(rhs.release())`. However, it is good practice that _every_ `new` has a matching `delete`, but this one will not have.

However, in the _extremely unlikely_ scenario where there's an exception after `new` but before `reset`, then you get another memory leak.